mixer/pkg/server/server_test.go

```diff
@@ -214,21 +237,79 @@ func TestErrors(t *testing.T) {
 					return nil, errors.New("BAD")
 				}
 			case 4:
-				pt.startMonitor = func(port uint16, enableProfiling bool) (*monitor, error) {
+				pt.startMonitor = func(port uint16, enableProfiling bool, lf listenFunc) (*monitor, error) {
 					return nil, errors.New("BAD")
 				}
 			case 5:
+				a.MonitoringPort = 1234
 				pt.listen = func(network string, address string) (net.Listener, error) {
-					return nil, errors.New("BAD")
+					if address != ":1234" {
+						return nil, errors.New("BAD")
+					}
+					return net.Listen(network, address)
+				}
+			case 6:
+				a.MonitoringPort = 1234
+				pt.listen = func(network string, address string) (net.Listener, error) {
+					if address == ":1234" {
+						return nil, errors.New("BAD")
+					}
+					return net.Listen(network, address)
+				}
+			case 7:
+				a.ConfigStoreURL = "http://abogusurl.com"
+			case 8:
+				pt.configLog = func(options *log.Options) error {
+					return errors.New("BAD")
+				}
+			case 9:
+				a.UseNewRuntime = true
+				pt.runtimeListen = func(rt *mixerRuntime2.Runtime) error {
+					return errors.New("BAD")
 				}
 			default:
 				return
 			}
 
-			s, err := newServer(a, pt)
+			s, err = newServer(a, pt)
 			if s != nil || err == nil {
 				t.Errorf("Got success, expecting error")
 			}
 		})
 	}
 }
+
+func TestMonitoringMux(t *testing.T) {
+	configStore, _ := storetest.SetupStoreForTest(globalCfg, serviceCfg)
+
+	a := DefaultArgs()
+	a.ConfigStore = configStore
+	a.MonitoringPort = 0
+	a.APIPort = 0
+	s, err := New(a)
+	if err != nil {
+		t.Fatalf("Got %v, expecting success", err)
+	}
+
+	r := &http.Request{}
+	r.Method = "GET"
+	r.URL, _ = url.Parse("http://localhost/version")
+	rw := responseWriter{}
+	s.monitor.monitoringServer.Handler.ServeHTTP(&rw, r)
```

Comment:
Can you add a comment on how this is testing things? It is not easy to parse and grok.